### DISCUSSION TITLE
Android - Geofence permission clarification

### DISCUSSION THREAD

**Discussion by Dipannita20** [shinyorg/shiny#1083] [Q&A] [answered]
Hi,

On Android 12 geofence registered successfully and location permission set to Allow All. But still geofence updates not triggering sometimes.

Are these below two permission required for geofence. https://github.com/shinyorg/samples/blob/main/Locations-Geofencing/Sample.Android/Properties/AndroidManifest.xml. 
```
	<uses-permission android:name="android.permission.START_FOREGROUND_SERVICES_FROM_BACKGROUND" />
	<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```
I have below permissions added and it used to work
`<uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
	<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
	<uses-permission android:name="android.permission.ACCESS_BACKGROUND_LOCATION" />`

Thanks in Advanced

**Comment by aritchie** [comment, marked as answer]
No those permissions are not required for the standard geofencing.  They are required if you are using the Gps/Geofence hybrid mechanism
